core/src/padding.rs

```diff
@@ -71,6 +73,18 @@ impl Padding {
     pub fn horizontal(self) -> u16 {
         self.left + self.right
     }
+
+    /// Constrains the padding to fit between the inner & outer [`Size`]
+    pub fn constrain(self, inner: Size, outer: Size) -> Self {
```

Comment:
The algorithm of constraining in min-max range usually named `clamp`, so
```suggestion
    pub fn clamp(self, inner: Size, outer: Size) -> Self {
```

Reply:
I thought of `clamp`, but since this isn't passing it a min / max `Size` I was unsure if it made sense

Reply:
> constrain a value..

Welp, guess constrain works too :P

Reply:
`clamp` just seems like it should only take `clamp(Self, Self)` is my only concern. There's no concept for `Padding.min(Size)` / `Padding.max(Size)`, so `Padding.clamp(Size, Size)` feels awkward

Reply:
I renamed it to `fit` in 7476663069572adec25161b46c26570f864f736f.